hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/workflow/report/ExtractionSummarizer.java

```diff
@@ -0,0 +1,76 @@
+package com.blackducksoftware.integration.hub.detect.workflow.report;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.blackducksoftware.integration.hub.detect.workflow.bomtool.BomToolEvaluation;
+import com.blackducksoftware.integration.hub.detect.workflow.codelocation.DetectCodeLocation;
+import com.blackducksoftware.integration.hub.detect.workflow.extraction.Extraction;
+import com.blackducksoftware.integration.hub.detect.workflow.extraction.Extraction.ExtractionResultType;
+
+public class ExtractionSummarizer extends BomToolEvaluationSummarizer {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public List<ExtractionSummaryData> summarize(final List<BomToolEvaluation> results, final Map<DetectCodeLocation, String> codeLocationNameMap) {
+        final Map<File, List<BomToolEvaluation>> byDirectory = groupByDirectory(results);
+
+        final List<ExtractionSummaryData> data = createSummaries(byDirectory, codeLocationNameMap);
+
+        final List<ExtractionSummaryData> sorted = data.stream()
+                .sorted((o1, o2) -> filesystemCompare(o1.directory, o2.directory))
+                .collect(Collectors.toList());
+
+        return sorted;
+    }
+
+    private List<ExtractionSummaryData> createSummaries(final Map<File, List<BomToolEvaluation>> byDirectory, final Map<DetectCodeLocation, String> codeLocationNameMap) {
+        final List<ExtractionSummaryData> datas = new ArrayList<>();
```

Comment:
Lots of nesting, can we break this out into more digestible methods and/or streams?

Reply:
I think I've improved it.